### PR TITLE
fix: fixed e2e test linter oversight + errors reported by static & Peter analysis

### PR DESCRIPTION
**Pull request by mzedel**

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -1,7 +1,6 @@
 import autoprefixer from 'autoprefixer';
 import { CleanWebpackPlugin } from 'clean-webpack-plugin';
 import CopyWebpackPlugin from 'copy-webpack-plugin';
-import { EsbuildPlugin } from 'esbuild-loader';
```
[the remark this answers is not in the document]
Comment (mzedel) [reply webpack.config.js:4]: exactly!

File: src/js/tracking.js [hunk 2]
```diff
@@ -77,8 +77,6 @@ class Tracker {
   pageview(data) {
     if (data) {
       this.currentPageView = data;
-    } else {
-      data = this.currentPageView;
```
[the remark this answers is not in the document]
Comment (mzedel) [reply src/js/tracking.js:81]: exactly!

File: src/js/components/common/dialogs/confirmdismisshelptips.js [hunk 3]
```diff
@@ -34,7 +34,7 @@ export const ConfirmDismissHelptips = () => {
       <DialogActions>
         <Button onClick={() => dispatch(setShowDismissOnboardingTipsDialog(false))}>Cancel</Button>
         <div style={{ flexGrow: 1 }} />
-        <Button variant="contained" color="secondary" onClick={() => dispatch(setOnboardingCanceled(true))}>
+        <Button variant="contained" color="secondary" onClick={() => dispatch(setOnboardingCanceled())}>
```
[the remark this answers is not in the document]
Comment (merlin-northern) [reply src/js/components/common/dialogs/confirmdismisshelptips.js:37]: I know, thats why I am sking about [./src/js/actions/onboardingActions.test.js:182](https://github.com/mendersoftware/gui/blob/master/src/js/actions/onboardingActions.test.js#L182)

File: src/js/components/deployments/deployment-wizard/softwaredevices.js [hunk 4]
```diff
@@ -62,8 +62,11 @@ export const getDevicesLink = ({ devices, group, hasFullFiltering, name }) => {
 };
 
 export const getDeploymentTargetText = ({ deployment, devicesById, idAttribute }) => {
-  const { devices = [], group = '', name = '', type = DEPLOYMENT_TYPES.software } = deployment;
-  const deviceList = (isUUID(name) && devicesById[name] ? [devicesById[name]] : Array.isArray(devices) ? devices : Object.values(devices)) ?? [];
+  const { devices = {}, group = '', name = '', type = DEPLOYMENT_TYPES.software } = deployment;
+  let deviceList = Array.isArray(devices) ? devices : Object.values(devices);
+  if (isUUID(name) && devicesById[name]) {
+    deviceList = [devicesById[name]];
+  }
```
Comment (merlin-northern) [inline src/js/components/deployments/deployment-wizard/softwaredevices.js:69]: delicious!

**Review by merlin-northern — APPROVED** [review]
approving not to block as there is only one outstanding comment. thank you for answering all -- as usual. :*

File: src/js/components/deployments/deployment-wizard/softwaredevices.js [hunk 5]
```diff
@@ -62,8 +62,8 @@ export const getDevicesLink = ({ devices, group, hasFullFiltering, name }) => {
 };
 
 export const getDeploymentTargetText = ({ deployment, devicesById, idAttribute }) => {
-  const { devices = [], group = '', name = '', type = DEPLOYMENT_TYPES.software } = deployment;
-  const deviceList = (isUUID(name) && devicesById[name] ? [devicesById[name]] : Array.isArray(devices) ? devices : Object.values(devices)) ?? [];
+  const { devices = {}, group = '', name = '', type = DEPLOYMENT_TYPES.software } = deployment;
+  const deviceList = isUUID(name) && devicesById[name] ? [devicesById[name]] : Array.isArray(devices) ? devices : Object.values(devices);
```
[the remark this answers is not in the document]
Comment (merlin-northern) [reply src/js/components/deployments/deployment-wizard/softwaredevices.js]: in time I will even be able to read it ;-) thanks!

File: src/js/components/common/dialogs/confirmdismisshelptips.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (mzedel) [reply src/js/components/common/dialogs/confirmdismisshelptips.js:37]: and kudos to you for the keen eye!!

**Pull request merged**